runtime/reconcilers/explore.go

```diff
@@ -132,6 +132,22 @@ func (r *ExploreReconciler) validateAndRewrite(ctx context.Context, self *runtim
 		return nil, nil, fmt.Errorf("parent metrics view %q is invalid", spec.MetricsView)
 	}
 
+	if mv.Parent != "" && spec.DefinedInMetricsView {
+		// during parsing, derived metrics view will not have the dimensions and measures resolved, so we need to get them from the valid metrics view spec here.
+		if len(spec.Dimensions) == 0 && spec.DimensionsSelector == nil {
+			for _, dim := range mv.Dimensions {
+				spec.Dimensions = append(spec.Dimensions, dim.Name)
+			}
+			spec.DimensionsSelector = nil
+		}
+		if len(spec.Measures) == 0 && spec.MeasuresSelector == nil {
+			for _, m := range mv.Measures {
+				spec.Measures = append(spec.Measures, m.Name)
+			}
+			spec.MeasuresSelector = nil
+		}
+	}
```

Comment:
Hmm, this feels like it should not be necessary. Can't this just be solved by setting `spec.DimensionsSelector` and `spec.MeasuresSelector` to `*` in the parser, which is already the default in the explore parser? See here: https://github.com/rilldata/rill/blob/96d18d51466b108ab3be0c33957b0fcc61e81dc5/runtime/parser/parse_partial_field_selector.go#L96-L99

Reply:
makes sense, can be done